core/src/Kokkos_CopyViews.hpp

```diff
@@ -1984,6 +2146,41 @@ void KOKKOS_INLINE_FUNCTION local_deep_copy(
 }
 //----------------------------------------------------------------------------
 template <class TeamType, class DT, class... DP, class ST, class... SP>
+void KOKKOS_INLINE_FUNCTION local_deep_copy_thread(
+    const TeamType& team, const View<DT, DP...>& dst,
+    const View<ST, SP...>& src,
+    std::enable_if_t<(unsigned(ViewTraits<DT, DP...>::rank) == 7 &&
+                      unsigned(ViewTraits<ST, SP...>::rank) == 7)>* = nullptr) {
+  if (dst.data() == nullptr) {
+    return;
+  }
+
+  const size_t N = dst.extent(0) * dst.extent(1) * dst.extent(2) *
+                   dst.extent(3) * dst.extent(4) * dst.extent(5) *
+                   dst.extent(6);
+
+  if (dst.span_is_contiguous() && src.span_is_contiguous()) {
+    local_deep_copy_contiguous_thread(team, dst, src);
+  } else {
+    Kokkos::parallel_for(Kokkos::ThreadVectorRange(team, N), [&](const int& i) {
+      int i0                          = i % dst.extent(0);
+      int itmp                        = i / dst.extent(0);
+      int i1                          = itmp % dst.extent(1);
+      itmp                            = itmp / dst.extent(1);
+      int i2                          = itmp % dst.extent(2);
+      itmp                            = itmp / dst.extent(2);
+      int i3                          = itmp % dst.extent(3);
+      itmp                            = itmp / dst.extent(3);
+      int i4                          = itmp % dst.extent(4);
+      itmp                            = itmp / dst.extent(4);
+      int i5                          = itmp % dst.extent(5);
+      int i6                          = itmp / dst.extent(5);
+      dst(i0, i1, i2, i3, i4, i5, i6) = src(i0, i1, i2, i3, i4, i5, i6);
+    });
```

Comment:
Is there a good reason not to use `ThreadVectorMDRange`?

Reply:
For this example, I think `MDRange` only goes up to rank 6, and `local_deep_copy` was already implemented this way.

Reply:
Right. That doesn't mean that we can't try to do better than the current implementation. 🙂